ci/github-script/bot.js

```
@@ -524,14 +564,20 @@ module.exports = async ({ github, context, core, dry }) => {
               'unmarked_as_duplicate',
             ].includes(event),
           )
-          .map(
-            ({ created_at, updated_at, committer, submitted_at }) =>
-              new Date(
-                updated_at ?? created_at ?? submitted_at ?? committer.date,
-              ),
-          )
+          .map((event) => {
+            // Timeline events are a union type; we access properties that may not exist on all types
+            // Safe because of fallback chain ending in 0
+            const anyEvent = /** @type {any} */ (event)
+            return new Date(
+              anyEvent.updated_at ??
+                anyEvent.created_at ??
+                anyEvent.submitted_at ??
+                anyEvent.committer?.date ??
+                0,
```

potentially we want to throw an error here?